optuna/visualization/matplotlib/_slice.py

```diff
@@ -73,7 +91,10 @@ def _get_slice_plot(study: Study, params: Optional[List[str]] = None) -> "Axes":
     plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
 
     # Prepare data.
-    obj_values = [t.value for t in trials]
+    if target is None:
+        obj_values = [cast(float, t.value) for t in trials]
```

Comment:
Note: `t.value` is guaranteed not to be `None` because they are filtered by `TrialState.COMPLETE`.